[2025-05-25T20:04:43.573+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Marketing_Data_Pipeline.dbt_run.dbt_test manual__2025-05-25T19:37:32.889912+00:00 [queued]>
[2025-05-25T20:04:43.699+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Marketing_Data_Pipeline.dbt_run.dbt_test manual__2025-05-25T19:37:32.889912+00:00 [queued]>
[2025-05-25T20:04:43.701+0000] {taskinstance.py:2170} INFO - Starting attempt 3 of 3
[2025-05-25T20:04:44.098+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): dbt_run.dbt_test> on 2025-05-25 19:37:32.889912+00:00
[2025-05-25T20:04:44.109+0000] {standard_task_runner.py:60} INFO - Started process 21343 to run task
[2025-05-25T20:04:44.116+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'Marketing_Data_Pipeline', 'dbt_run.dbt_test', 'manual__2025-05-25T19:37:32.889912+00:00', '--job-id', '161', '--raw', '--subdir', 'DAGS_FOLDER/Marketing_Data_Pipeline.py', '--cfg-path', '/tmp/tmpi6vwdw2e']
[2025-05-25T20:04:44.122+0000] {standard_task_runner.py:88} INFO - Job 161: Subtask dbt_run.dbt_test
[2025-05-25T20:04:44.377+0000] {task_command.py:423} INFO - Running <TaskInstance: Marketing_Data_Pipeline.dbt_run.dbt_test manual__2025-05-25T19:37:32.889912+00:00 [running]> on host fe289fe396bd
[2025-05-25T20:04:44.863+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='yashbhawsar872@gmail.com' AIRFLOW_CTX_DAG_OWNER='Yash' AIRFLOW_CTX_DAG_ID='Marketing_Data_Pipeline' AIRFLOW_CTX_TASK_ID='dbt_run.dbt_test' AIRFLOW_CTX_EXECUTION_DATE='2025-05-25T19:37:32.889912+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-05-25T19:37:32.889912+00:00'
[2025-05-25T20:04:44.867+0000] {Marketing_Data_Pipeline.py:39} INFO - ‚ñ∂Ô∏è Starting task: Run all dbt tests
[2025-05-25T20:04:44.892+0000] {utils.py:93} INFO - Running dbt tests: cd /opt/dbt/marketing_pipeline && dbt test 
[2025-05-25T20:04:44.900+0000] {utils.py:65} INFO - Running dbt command: cd /opt/dbt/marketing_pipeline && dbt test 
[2025-05-25T20:05:31.566+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/Marketing_Data_Pipeline.py", line 40, in _wrapped
    fn(**kwargs)
  File "/opt/airflow/scripts/utils.py", line 95, in run_dbt_test
    run_dbt(cmd)
  File "/opt/airflow/scripts/utils.py", line 75, in run_dbt
    full_log = out + err
TypeError: can only concatenate str (not "NoneType") to str
[2025-05-25T20:05:31.719+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=Marketing_Data_Pipeline, task_id=dbt_run.dbt_test, execution_date=20250525T193732, start_date=20250525T200443, end_date=20250525T200531
[2025-05-25T20:05:31.805+0000] {email_notification.py:58} WARNING - üìß Sending generic failure email [Airflow] Failure in Marketing_Data_Pipeline.dbt_run.dbt_test (Attempt 4/3)
[2025-05-25T20:05:31.903+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.10/site-packages/***/utils/email.py:154 RemovedInAirflow3Warning: Fetching SMTP credentials from configuration variables will be deprecated in a future release. Please set credentials using a connection instead.
[2025-05-25T20:05:31.908+0000] {email.py:270} INFO - Email alerting: attempt 1
[2025-05-25T20:05:33.411+0000] {email.py:281} INFO - Sent an alert email to ['yashbhawsar872@gmail.com']
[2025-05-25T20:05:39.912+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 161 for task dbt_run.dbt_test (can only concatenate str (not "NoneType") to str; 21343)
[2025-05-25T20:05:40.148+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-05-25T20:05:40.256+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
